### PR TITLE
add geolocation button

### PR DESCRIPTION
**Pull request by simonseyock**
This adds a simple geolocation tool that is toggable.

### CODE REVIEW

File: src/view/button/Geolocation.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    },
+
+    updateAccuracyFeature: function (geom) {
+        var me = this;
+        if (geom) {
+            if (!me.accuracyFeature) {
+                me.accuracyFeature = new ol.Feature();
+                me.geolocationVectorLayer.getSource().addFeature(me.accuracyFeature);
+            }
+            me.accuracyFeature.setGeometry(geom)
```
Comment (annarieger) [inline src/view/button/Geolocation.js:183]: missing semicolon

File: src/view/button/Geolocation.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        }
+    },
+
+    /**
+     * Called when the button is toggled, this method ensures that everything
+     * is cleaned up when unpressed, and that measuring can start when pressed.
+     *
+     * @param {Ext.Button} btn The measure button itself.
+     * @param {Boolean} pressed Whether the button is now pressed or not.
+     */
```
Comment (annarieger) [inline src/view/button/Geolocation.js:198]: Misleading copied docs :)

File: src/view/button/Geolocation.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+        }
+    },
+
+    /**
+     * Creates a correctly setup vector layer on which the draw interaction will
+     * work.
+     *
+     * @return {ol.layer.Vector} The layer on which the draw interaction will
+     *     work.
+     */
```
Comment (annarieger) [inline src/view/button/Geolocation.js:214]: Also misleading copied docs :)

**Review by annarieger — COMMENTED** [review]
Very nice addition! Please address my notes before merge 🍍

File: src/view/button/Geolocation.js (hunk 2 above)
Comment (simonseyock) [reply src/view/button/Geolocation.js:198]: yeah. fixed that already :sweat_smile:

**Pull request merged**